input_data_scanner.py

```diff
@@ -0,0 +1,59 @@
+import glob
+import queue
+
+class FileScanner:
+    """
+      A class to scan a directory for new files with a specified file extension
+      and add their file paths to a queue.
+
+      Attributes:
+      - existing_files (set): A set to hold the paths of existing files.
+      - cv_queue (queue.Queue): A queue to hold the paths of new files.
+
+      Methods:
+      - scan(): Scans the directory for new files and adds their file paths to the queue.
+      """
+    def __init__(self,cv_queue):
+        """
+               Initializes a new instance of the FileScanner class.
+
+               Args:
+               - cv_queue (queue.Queue): A queue to hold the paths of new files.
+        """
+        # Create a set to hold the paths of existing files
+        self.existing_files = set()
+        # Create a queue to hold the paths of new files
+        self.cv_queue = queue.Queue()
+
+    def scan(self):
+        """
+              Scans the directory for new files and adds their file paths to the queue.
+        """
+        # Get a list of paths of all files with the desired file extension
+        try:
+            filepaths = set(glob.glob('C:/Users/LAP-10/Documents/input_data/*.pdf'))
```

Comment:
We can not use a fixed absolute path here. This can only run on a specific system for now.

Reply:
Done